src/gui/src/staGui.cpp

```diff
@@ -213,24 +218,80 @@ void TimingPathsModel::populateModel(bool setup_hold, int path_count)
 }
 
 bool TimingPathsModel::populatePaths(bool get_max,
-                                     int path_count,
-                                     bool clockExpanded)
+                                     int path_count)
 {
   // On lines of DataBaseHandler
   QApplication::setOverrideCursor(Qt::WaitCursor);
-  sta_->ensureGraph();
-  sta_->searchPreamble();
 
-  auto sta_state = sta_->search();
+  TimingPath::buildPaths(sta_,
+                         get_max,
+                         false, // unconstrained
+                         path_count,
+                         {}, // to
+                         {}, // through
+                         {}, // from
+                         true,
+                         timing_paths_);
+
+  QApplication::restoreOverrideCursor();
+  return true;
+}
+
+/////////
+
+void TimingPath::buildPaths(sta::dbSta* sta,
+                            bool get_max,
+                            bool include_unconstrained,
+                            int path_count,
+                            const std::set<sta::Pin*>& from,
+                            const std::set<sta::Pin*>& thrus,
+                            const std::set<sta::Pin*>& to,
+                            bool include_capture,
+                            std::vector<std::unique_ptr<TimingPath>>& paths)
+{
+  sta->ensureGraph();
+  sta->searchPreamble();
+
+  sta::ExceptionFrom* e_from = nullptr;
+  if (!from.empty()) {
+    sta::PinSet* pins = new sta::PinSet;
+    pins->insert(from.begin(), from.end());
+    e_from = sta->makeExceptionFrom(pins,
+                                    nullptr,
+                                    nullptr,
+                                    sta::RiseFallBoth::riseFall());
```

Comment:
Is the ownership semantic clear - I am wondering who free pins & e_from?  Likewise below.

Reply:
Based on the comments in the Sta.hh header:
https://github.com/The-OpenROAD-Project/OpenSTA/blob/e826e4e27e5543661719fa4e46cd6f31dffadee0/include/sta/Sta.hh#L827
Search owns those and deletes them.